### PR TITLE
[WIP][vcpkg]Update openssl from 1.0.2 to 1.1.0

### PR DESCRIPTION
**Pull request by Alisayyy-Yang**

### CODE REVIEW

**Comment by KindDragon** [comment]
Related issue #218

**Comment by alexkaratarakis** [comment]
- Added patches for `websocketpp` and `cpprestsdk` for compatibility with openssl 1.1.0
- Added wincrypt, winsock2 ports and depend on them; fixes linking issues.

 (Independent changes were added on master, so I rebased this on top to keep the PR page clean)

File: ports/openssl/portfile.cmake [hunk 1]
```diff
@@ -97,46 +84,46 @@ set(SOURCE_PATH_DEBUG ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/openssl-${
...
+    ${CURRENT_PACKAGES_DIR}/debug/bin/ossl_static.pdb
+    ${CURRENT_PACKAGES_DIR}/bin/ossl_static.pdb
+    ${CURRENT_PACKAGES_DIR}/debug/openssl.cnf.dist
+    ${CURRENT_PACKAGES_DIR}/openssl.cnf.dist
     ${CURRENT_PACKAGES_DIR}/debug/openssl.cnf
     ${CURRENT_PACKAGES_DIR}/openssl.cnf
 )
 
-file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools/${PORT}/)
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/openssl.exe ${CURRENT_PACKAGES_DIR}/tools/${PORT}/openssl.exe)
```
Comment (Mixaill) [inline ports/openssl/portfile.cmake]: regression, see #1867

Comment (alexkaratarakis) [reply ports/openssl/portfile.cmake]: Fixed with 14e0217, thanks!

**Comment by sipsorcery** [comment]
Will this PR mean vcpkg can only install openssl 1.1.0? 

If so that's going to be a breaking change for a lot of other packages that depend on openssl, for example qt. The API and build artefacts changed a lot going from 1.0.x to 1.1.0. Would it be feasible to leave the current openssl package at version 1.0.x and create a new separate one for 1.1.0?

**Comment by alexkaratarakis** [comment]
Before merging this, we would want most (ideally all) ports in vcpkg that depend on openssl to work with the new version. Here is a list of known failures:
- [x] cpprestsdk
- [ ] freerdp
- [ ] grpc
- [ ] libevent
- [ ] libpq
- [ ] libssh
- [ ] libtorrent
- [ ] libwebsockets
- [ ] opusfile
- [ ] paho-mqtt
- [x] uwebsockets
- [ ] thrift
- [ ] wt

(Checked ones have been fixed). 
At that point we will re-evaluate our options in terms of what to do with the version. `openssl`'s omnipresence may require us to do special handling to avoid breakages.

**Pull request closed without merging**